.github/workflows/build-design-system.yml

```
@@ -16,19 +16,46 @@ jobs:
     # Step 1: Checkout the repository
     - name: Checkout repository
       uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }} # Checkout the branch for the PR
 
-    # Step 2: Set up Bun
+    # Step 2: Cache Bun dependencies
+    - name: Cache Bun dependencies
+      id: cache-deps
+      uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)